Core/Core/Store/Store.swift

```diff
@@ -119,6 +128,41 @@ public class Store<U: UseCase>: NSObject, NSFetchedResultsControllerDelegate {
         }
     }
 
+    @objc
+    public func controllerWillChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
+        changes = []
+    }
+
+    @objc
+    public func controller(
+        _ controller: NSFetchedResultsController<NSFetchRequestResult>,
+        didChange sectionInfo: NSFetchedResultsSectionInfo,
+        atSectionIndex sectionIndex: Int,
+        for type: NSFetchedResultsChangeType
+    ) {
+        changes.append(type == .delete ? .deleteSection(sectionIndex) : .insertSection(sectionIndex))
+    }
+
+    @objc
+    public func controller(
+        _ controller: NSFetchedResultsController<NSFetchRequestResult>,
+        didChange anObject: Any,
+        at indexPath: IndexPath?,
+        for type: NSFetchedResultsChangeType,
+        newIndexPath: IndexPath?
+    ) {
+        if type == .insert, let index = newIndexPath {
+            changes.append(.insertRow(index))
+        } else if type == .delete, let index = indexPath {
+            changes.append(.deleteRow(index))
+        } else if type == .update, let index = indexPath {
+            changes.append(.updateRow(index))
```

Comment:
it's not so much the performance benefit, as the animation for the new cell, instead of janky reloading of everything. It looks particularly bad when you have a screen full of audio comments.